neqo-http3/src/stream_type_reader.rs

```diff
@@ -69,16 +65,13 @@ impl RecvStream for NewStreamTypeReader {
     }
 
     fn receive(&mut self, conn: &mut Connection) -> Res<ReceiveOutput> {
-        let stream_type = self.get_type(conn, self.stream_id);
-        if let Some(t) = stream_type {
-            Ok(ReceiveOutput::NewStream(t))
-        } else {
-            Ok(ReceiveOutput::NoOutput)
-        }
+        Ok(self
+            .get_type(conn)
+            .map_or(ReceiveOutput::NoOutput, |t| ReceiveOutput::NewStream(t)))
     }
 
     fn done(&self) -> bool {
-        self.fin
+        *self == NewStreamTypeReader::Done
```

Comment:
You can get the same result without implementing PartialEq on the incremental reader using: `matches!(self, NewStreamTypeReader::Done)`.